*** linux-3.18.7/arch/arm/mach-bcm2708/bcm2708.c	2015-02-18 11:25:24.800413444 +0100
--- linux-3.18.7.patched/arch/arm/mach-bcm2708/bcm2708.c	2015-02-17 20:41:32.519408193 +0100
***************
*** 938,948 ****
--- 938,950 ----
  	system_rev = boardrev;
  	system_serial_low = serial;
  
+ #if 0
  #ifdef CONFIG_BCM2708_SPIDEV
  	if (!use_dt)
  	    spi_register_board_info(bcm2708_spi_devices,
  				    ARRAY_SIZE(bcm2708_spi_devices));
  #endif
+ #endif
  extern void babelbee_init(void);
  	babelbee_init();
  }
